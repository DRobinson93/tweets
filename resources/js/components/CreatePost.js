import React from 'react';
import AlertMessage from './presontational/AlertMessage';
import axios from 'axios';

class CreatePost extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            value: '',
            alert: {}
        };
    }
    handleChange = (event) => {
        this.setState({value: event.target.value});
    };
    handleSubmit = () => {
        this.setState({alert: {}});
        if(this.state.value === ""){
            return this.setState({
                alert: {text:'Tweet can not be blank', type:'info'}
            });
        }
        axios.post('/posts', {
            value: this.state.value
        })
            .then(response => {
                this.props.addToParentPostArr(response.data);
                this.setState({
                    alert: {text:'Tweet added', type:'success'},
                    value : ''
                });
            })
            .catch(error => {
                this.setState({
                    alert: {text:error, type:'danger'}
                });
            });
    };
    render() {
        return (
            <div className="form-inline align-items-center">
                <div className="form-row col-12">
                    <label className="sr-only" htmlFor="inlineFormInput">New Tweet</label>
                    <textarea onChange={this.handleChange} className="form-control mb-2 col-11" id="inlineFormInput" placeholder="What's going on?"/>
                    <button onClick={this.handleSubmit} type="button" className="btn btn-primary mb-2 rounded col-1">Tweet</button>
                </div>
                <AlertMessage show="true" type={this.state.alert.type} text={this.state.alert.text}/>
            </div>
        );
    }
}

export default CreatePost;
