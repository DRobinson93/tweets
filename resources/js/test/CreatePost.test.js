import React from 'react';
import CreatePost from '../components/CreatePost';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react'

test('Empty tweet gives an alert error', () => {
    const component = renderer.create(
        <CreatePost />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually click submit btn
    component.root.findByType('button').props.onClick();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

const setup = () => {
    const utils = render(<CreatePost />)
    const input = utils.getByLabelText('New Tweet')
    return {
        input,
        ...utils,
    }
}

test('Input value is not changed', () => {
    const { input } = setup()
    fireEvent.change(input, { target: { value: '23' } });
    expect(input.value).toBe('23')
})
