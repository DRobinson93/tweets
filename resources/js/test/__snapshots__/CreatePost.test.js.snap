// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty tweet gives an alert error 1`] = `
<div
  className="form-inline align-items-center"
>
  <div
    className="form-row col-12"
  >
    <label
      className="sr-only"
      htmlFor="inlineFormInput"
    >
      New Tweet
    </label>
    <textarea
      className="form-control mb-2 col-11"
      id="inlineFormInput"
      onChange={[Function]}
      placeholder="What's going on?"
    />
    <button
      className="btn btn-primary mb-2 rounded col-1"
      onClick={[Function]}
      type="button"
    >
      Tweet
    </button>
  </div>
</div>
`;

exports[`Empty tweet gives an alert error 2`] = `
<div
  className="form-inline align-items-center"
>
  <div
    className="form-row col-12"
  >
    <label
      className="sr-only"
      htmlFor="inlineFormInput"
    >
      New Tweet
    </label>
    <textarea
      className="form-control mb-2 col-11"
      id="inlineFormInput"
      onChange={[Function]}
      placeholder="What's going on?"
    />
    <button
      className="btn btn-primary mb-2 rounded col-1"
      onClick={[Function]}
      type="button"
    >
      Tweet
    </button>
  </div>
  <div
    className="fade alert alert-info show"
    fade="false"
    role="alert"
  >
    Tweet can not be blank
  </div>
</div>
`;
